common/src/network.rs

```diff
@@ -99,10 +100,31 @@ impl Network {
         }
     }
 
+    /// Return a list of services that nakamoto need
+    /// in the DNS resolving
+    fn supported_services(&self) -> vec::Vec<ServiceFlags> {
+        vec![ServiceFlags::NETWORK, ServiceFlags::COMPACT_FILTERS]
+    }
+
+    /// Return a list of flag in string slices form
+    fn service_flags(&self) -> vec::Vec<String> {
+        let services = self.supported_services();
+        services
+            .iter()
+            .map(|service| match *service {
+                //FIXME(vincenzopalazzo): The COMPACT_FILTER is it x64? Check in bitcoin core.
+                ServiceFlags::NETWORK | ServiceFlags::COMPACT_FILTERS => {
+                    format!("x{}", service.as_u64())
+                }
+                _ => panic!("Unsupported feature!"),
+            })
+            .collect()
+    }
```

Comment:
I implemented a similar solution that you suggested here, I don't know if I got all the points.

with the actual solution we can avoid checking how many peers we have resolved because if we fail in a specific resolution by services flags, we will make a generic resolution (without services flags).